### ISSUE TITLE
Provide HSTORE example

### ISSUE THREAD

**Issue by ryanbyyc** [go-pg/pg#248]
Struct 

```sql
CREATE TABLE IF NOT EXISTS Mytable (
    id bigserial primary key,
    records hstore NOT NULL DEFAULT hstore(array[]::varchar[])
);
```

```go
type Mytable struct {
	ID           int64
	Records       [][]string `pg:",array"`
}
```

Results:
`pg: can't parse string slice: "\"foo\"=>\"bar\""`

**Comment by vmihailenco** [comment]
Currently hstore is not supported. The idea is that you will use jsonb instead :)

I can try to add hstore support for maps, e.g. `pg.Hstore(map[string]string)` or `Records map[string]string pg:",hstore"`.

What `hstore(array[]::varchar[])` is supposed to do? I can'f find any hints in docs...

**Comment by ryanbyyc** [comment]
Apologies @vmihailenco, the DEFAULT in the declaration was an attempt to set the default to be HSTORE string. For this example, let's assume records is simply defined as `records hstore` in the table create schema. I was trying to find a way to avoid doing this:
`update mytable set records = COALESCE(records, hstore('')) || hstore('foo', 'bar') WHERE id = 1;`

I had searched the repo for hstore so I had assumed it was supported, but it would be grand if it was. Since Postgres treats hstore data as a string, I'd love support!

As you said, I think this is the way to go:
```
type Mytable struct {
    ID           int64
    Records map[string]string pg:",hstore"
}
```

**Comment by ryanbyyc** [comment]
Pulled and initial tests passed with flying colors, looks like it all works wonderfully. Thanks for the support! Closing issue.

**Comment by vmihailenco** [comment]
Thanks for testing. Released as v4.6.0.
